Dockerfile

```diff
@@ -48,19 +47,17 @@ RUN touch crates/pathfinder/src/build.rs
 RUN touch crates/stark_curve/src/lib.rs
 RUN touch crates/stark_hash/src/lib.rs
 
-RUN RUSTFLAGS='-L/usr/lib -Ctarget-feature=-crt-static' cargo build --release -p pathfinder
+RUN cargo build --release -p pathfinder
 
 #######################################
 # Stage 2: Build the Python libraries #
 #######################################
-FROM python:3.8-alpine AS python-builder
+FROM python:3.8-slim AS python-builder
 
-RUN apk upgrade --no-cache
-RUN apk add --no-cache gcc musl-dev gmp-dev g++
+RUN apt update && DEBIAN_FRONTEND=noninteractive apt install -y libgmp-dev gcc && rm -rf /var/lib/apt/lists/*
 
 WORKDIR /usr/share/pathfinder
 COPY py py
-RUN python3 -m pip install --upgrade pip
```

Comment:
Why would you remove this?

Reply:
Because this leads to issues when we're copying `site-packages` over to the final image: since the image already has pip installed, we just copy over files from a new version, while leaving files from the old `pip` version there. This also makes the resulting image _slightly_ larger.

And we don't seem to need the upgrade anyway, pip is at 22.0.4 on the base image.

Reply:
Why do we actually copy those site-packages things over instead of just leaving them at the runner....? To delete the test files which hopefully are not needed?

This will create at least a difference in between the two ways of compiling from source. If virtualenv was copied it would be safer to upgrade pip in that, then just copy the virtualenv over.

Reply:
> Why do we actually copy those site-packages things over instead of just leaving them at the runner....?

Size optimization I guess -- not only the test files, but we also don't need `gcc` and `libgmp-dev` on the runner.

> This will create at least a difference in between the two ways of compiling from source.

What do you mean here?

Reply:
>> This will create at least a difference in between the two ways of compiling from source.
> What do you mean here?

Meant the dockerfile as a way to compile from source, the README has this pip upgrade. I added it originally because with venv it should be always ok.

Reply:
Ah, OK. But why exactly do we need it in the Dockerfile, where we know _exactly_ that the version of pip that comes with the base image is OK? Seems like an unnecessary step to me.

Reply:
Yes it would seem unnecessary if we know the version.